sdcm/cluster.py

```diff
@@ -1302,12 +1302,12 @@ def _report_housekeeping_uuid(self, verbose=False):
             else:
                 self.remoter.sudo('touch %s' % mark_path, verbose=verbose, user='scylla')
 
-    def wait_db_up(self, verbose=True, timeout=3600):
+    def wait_db_up(self, verbose=True, timeout=3600, stop_event: threading.Event | None = None):
         text = None
         if verbose:
             text = '%s: Waiting for DB services to be up' % self
-        wait.wait_for(func=self.db_up, step=60, text=text, timeout=timeout, throw_exc=True)
-        self.db_init_finished = True
+        res = wait.wait_for(func=self.db_up, step=60, text=text, timeout=timeout, throw_exc=True, stop_event=stop_event)
+        self.db_init_finished = bool(res)
```

Comment:
since `throw_exc=True` I would assume that in case of failure `wait_for` won't return.
so why changing this to look at the result ?

Reply:
This was as example how it could be used. Removed this changes